src/pages/resultsView/download/DownloadUtils.ts

```diff
@@ -321,6 +322,79 @@ export function generateOtherMolecularProfileDownloadData(
         : [];
 }
 
+export function generateGenericAssayProfileData(
+    molecularProfileIds: string[],
+    unfilteredCaseAggregatedData: CaseAggregatedData<GenericAssayData>
+): { [key: string]: GenericAssayData[] } {
+    const sampleFilter = (data: GenericAssayData) => {
+        return molecularProfileIds.includes(data.molecularProfileId);
+    };
+
+    // generate GenericAssay profile data by key
+    // key => stableId + uniqueSampleKey
+    const sampleDataByStableId: { [key: string]: GenericAssayData[] } = {};
+
+    _.values(unfilteredCaseAggregatedData.samples).forEach(alterations => {
+        alterations.forEach(alteration => {
+            const key = `${alteration.stableId}_${alteration.uniqueSampleKey}`;
+            sampleDataByStableId[key] = sampleDataByStableId[key] || [];
+
+            // alteration is filtered out if filter function returns false
+            if (sampleFilter(alteration)) {
+                sampleDataByStableId[key].push(alteration);
+            }
+        });
+    });
+    return sampleDataByStableId;
+}
+
+export function generateGenericAssayProfileDownloadData(
+    sampleGenericAssayDataByStableId: { [key: string]: GenericAssayData[] },
+    samples: Sample[] = [],
+    stableIds: string[] = []
+): string[][] {
+    if (_.isEmpty(sampleGenericAssayDataByStableId)) {
+        return [];
+    } else {
+        // we need the sample index for better performance
+        const sampleIndex = _.keyBy(samples, 'uniqueSampleKey');
+        const sampleKeys = samples.map(sample => sample.uniqueSampleKey);
+
+        // generate row data (keyed by uniqueSampleKey)
+        const rows = generateGenericAssayDownloadFileRows(
```

Comment:
lets call this rowsByUniqueSampleKey

Reply:
@alisman Do you think `generateGenericAssayRowsByUniqueSampleKey` is good? There are some parameters in this function are specific to `generic assay data`